gl.h

```diff
@@ -0,0 +1,575 @@
+	float camZ;
+
+	float pitch;
+	float yaw;
+	float roll;
+} myCam;
+
+float threshold = 1.0;
+/**********************************************************
+ * < END OF GLOBALS >
+ *********************************************************/
+#define CAM_INCREMENT 0.05
+#define STEP_INCREMENT 0.05
+
+// Update state based on keyboard
+bool processUserInputs(bool & running)
+{
+	SDL_Event e;
+	while(SDL_PollEvent(&e)) 
+	{
+		if(e.type == SDL_QUIT) 
+		{
+			running = false;
+		}
+		if(e.type == SDL_KEYDOWN && e.key.keysym.sym == 'q') 
+		{
+			running = false;
+			break;
+		}
+		if(e.type == SDL_KEYDOWN && e.key.keysym.sym == 't') 
+		{
+			threshold -= 0.01;
+			if(threshold < 0)
+			{
+				threshold = 0;
+			}
+			break;
+		}
+		if(e.type == SDL_KEYDOWN && e.key.keysym.sym == 'y') 
+		{
+			threshold += 0.01;
+			if(threshold > 1.0)
+			{
+				threshold = 1.0;
+			}
+			break;
+		}
+
+        if(e.type == SDL_MOUSEMOTION)
+        {
+            int cur = SDL_ShowCursor(SDL_QUERY);
+            if(cur == SDL_DISABLE)
+            {
+                double mouseX = e.motion.xrel;
+                double mouseY = e.motion.yrel;
+                myCam.yaw   -= (mouseX * CAM_INCREMENT);
+                myCam.pitch -= (mouseY * CAM_INCREMENT);
+            }
+        }
+        if(e.type == SDL_MOUSEBUTTONDOWN)
+        {
+            int cur = SDL_ShowCursor(SDL_QUERY);
+            if(cur == SDL_DISABLE)
+            {
+                SDL_ShowCursor(SDL_ENABLE);
+                SDL_SetRelativeMouseMode(SDL_FALSE);
+            }
+            else
+            {
+                SDL_ShowCursor(SDL_DISABLE);
+                SDL_SetRelativeMouseMode(SDL_TRUE);
+            }
+        }
+
+        // Translation
+        if((e.key.keysym.sym == 'w' && e.type == SDL_KEYDOWN))
+        {
+
+            myCam.camZ -= (cos((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+            myCam.camX -= (sin((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+        }
+        if(e.key.keysym.sym == 's' && e.type == SDL_KEYDOWN)
+        {
+            myCam.camZ += (cos((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+            myCam.camX += (sin((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+        }
+        if(e.key.keysym.sym == 'a' && e.type == SDL_KEYDOWN)
+        {
+            myCam.camX -= (cos((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+            myCam.camZ += (sin((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+        }
+        if(e.key.keysym.sym == 'd' && e.type == SDL_KEYDOWN)
+        {
+			myCam.camX += (cos((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+            myCam.camZ -= (sin((myCam.yaw / 180.0) * M_PI)) * STEP_INCREMENT;
+        }
+	}
+}
+
+
+struct bmpRGB
+{
+  unsigned char b;
+  unsigned char g; 
+  unsigned char r;
+};
+
+// The portion of the Bitmap header we want to read
+struct bmpLayout
+{
+  int offset;
+  int headerSize;
+  int width;
+  int height;
+  unsigned short colorPlanes;
+  unsigned short bpp;
```

Comment:
are there other variables that we could be added here for the bmpLayout or this all of the variables?